<p>Avoid <a href="https://sling.apache.org/apidocs/sling10/org/apache/sling/jcr/api/SlingRepository.html#loginAdministrative-java.lang.String-">Sling
    Repository</a> loginAdministrative and
    <a href="https://sling.apache.org/apidocs/sling11/org/apache/sling/api/resource/ResourceResolverFactory.html#getAdministrativeResourceResolver-java.util.Map-">ResourceResolverFactory</a>
    getAdministrativeResourceResolver methods. They are deprecated and a security risk.</p>

<h2>Noncompliant Code Example</h2>
<pre>
@Model(adaptables = Resource.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
class SlingModelClass {

    @Reference
    private SlingRepository repository;

    public void aMethod() {
        Session session = null;
        try {
            session = repository.loginAdministrative(null);
        } catch (final LoginException | RepositoryException ex) {
            System.out.error("Unable to open repository session", ex);
        }
    }

}
</pre><h2>Compliant Solution</h2>
<pre>
@Model(adaptables = Resource.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
class SlingModelClass {

    @Reference
    private SlingRepository repository;

    public void aMethod() {
        Session session = null;
        try {
            session = repository.loginService(null, null);
        } catch (final LoginException | RepositoryException ex) {
            System.out.error("Unable to open repository session", ex);
        } finally {
            if (session != null && session.isLive()) {
                session.logout();
            }
        }

    }

}
</pre>