<p>When writing tests with aem-mocks avoid mocking the implementation of a OSGi component, but
    rather create a stub implementation or mock the interface of the component instead if really needed.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
@ExtendWith({AemContextExtension.class, MockitoExtension.class})
class TestClass {

    @Mock
    private DummyServiceImpl dummyServiceImpl;

    @Test
    void testDummmy() {
        ...
    }
}
</pre><h2>Compliant Solution</h2>
<pre>
@ExtendWith(AemContextExtension.class)
class TestClass {

    @Test
    void testDummmy() {
        AemContext aemContext = new AemContextBuilder().build();
        aemContext.registerService(DummyService.class, new DummyServiceMock());
        ...
    }
}
</pre>

<p>or</p>

<pre>
@ExtendWith({AemContextExtension.class, MockitoExtension.class})
class TestClass {

    @Mock
    private DummyService dummyService;

    @Test
    void testDummmy() {
        ...
    }
}
</pre>
