<p>When writing tests with aem-mocks it is preferred not to mock resource handling objects, such as ResourceResolver or Session, as they are already provided
    in the test context. You can also specify the type of the ResourceResolver implementation you wish to use from the multiple ones provided by the
    aem-mocks library, depending on the level of functionality you need for the test environment. </p>

<h2>Noncompliant Code Example</h2>
<pre>
@ExtendWith({AemContextExtension.class, MockitoExtension.class})
class TestClass {

    @Mock
    private ResourceResolver resourceResolver;

    @Mock
    private Session session;

    @Test
    void testDummmy() {
        ...
    }
}
</pre><h2>Compliant Solution</h2>
<pre>
@ExtendWith(AemContextExtension.class)
class TestClass {

    @Test
    void testDummmy() {
        AemContext aemContext = new AemContextBuilder(ResourceResolverType.JCR_MOCK).build();
        ResourceResolver resourceResolver = aemContext.resourceResolver();
        Session session = resourceResolver.adaptTo(Session.class);
        ...
    }
}
</pre>
