<p>Use the new Sling servlet OSGi DS 1.4 (R7) component property type annotations that come with org.apache.sling.servlets.annotations:
    @SlingServletResourceTypes and @SlingServletPaths.
</p>
<p>
Note - make sure you have the
<a href="https://github.com/apache/sling-org-apache-sling-servlets-annotations">Apache Sling Servlet Annotations</a>
Maven dependency added and that the maven-bundle-plugin dependency has been to 4.1.0+ version for the new standard to be enabled.
More information available in the
<a href="https://sling.apache.org/documentation/the-sling-engine/servlets.html">documentation</a>
</p>
<h2>
    Noncompliant Code Example
</h2>
<pre>
@Component(service = Servlet.class, immediate = true,
  property = { ServletResolverConstants.SLING_SERVLET_RESOURCE_TYPES + "=" + "resourceType",
    ServletResolverConstants.SLING_SERVLET_SELECTORS + "=" + "selector" })
public class SlingServlet extends SlingSafeMethodsServlet {}
</pre><h2>Compliant Solution</h2>
<pre>
@Component(service = Servlet.class, immediate = true)
@SlingServletResourceTypes(resourceTypes = "resourceType", selectors = "selector")
public class SlingServlet extends SlingSafeMethodsServlet {}
</pre>