<p>Whenever possible prefer using @SlingServletResourceTypes over @SlingServletPaths. If you need to use paths, make sure that the path you use is allowed in
    Apache Sling Servlet/Script Resolver configuration.

    Make use of constants for defining servlet properties in @SlingServletResourceTypes annotation to improve code readability. </p><h2>
    Noncompliant Code Example </h2>
<pre>
@Component(service = Servlet.class)
@SlingServletPaths("/bin/custom-servlet")
public class UserReportServlet extends SlingSafeMethodsServlet {}
</pre><h2>Compliant Solution</h2>
<pre>
@Component(service = Servlet.class, immediate = true)
@SlingServletResourceTypes(resourceTypes = "resourceType", selectors = "selector")
public class SlingServlet extends SlingSafeMethodsServlet {}
</pre>